### PR TITLE
[bitnami/kafka] Allow Kafka provisioning to be assigned a service account

### PR DESCRIPTION
**Pull request by cleverhu**
### Description of the change
Allow kakfa provisioning to be assigned a service account.

### Benefits
User can config the service account for kakfa provisioning  and zookeeper.

### Applicable issues

  - fixes https://github.com/bitnami/charts/issues/13080

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kafka/values.yaml [hunk 1]
```diff
@@ -1717,6 +1732,24 @@ zookeeper:
...
+    ## If not set and create is true, a name is generated using the common.names.fullname template
+    ##
+    name: ""
+    ## @param zookeeper.serviceAccount.automountServiceAccountToken Allows auto mount of ServiceAccountToken on the serviceAccount created
+    ## Can be set to false if pods using this serviceAccount do not need to use K8s API
+    ##
+    automountServiceAccountToken: true
+    ## @param zookeeper.serviceAccount.annotations Additional custom annotations for the ServiceAccount
+    ##
+    annotations: {}
```
Comment (jotamartos) [inline bitnami/kafka/values.yaml:1752]: These are the default values so this shouldn't be necessary

https://github.com/bitnami/charts/blob/main/bitnami/zookeeper/values.yaml#L564

File: bitnami/kafka/templates/_helpers.tpl [hunk 2]
```diff
@@ -279,6 +279,17 @@ Returns the secret name for the Kafka Provisioning client
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create the name of the service account to use for the Kafka Provisioning client 
+*/}}
+{{- define "kafka.client.serviceAccountName" -}}
```
Comment (jotamartos) [inline bitnami/kafka/templates/_helpers.tpl:285]: Isn't it clearer if we use "kafka.provisioning.serviceAccountName"?

File: bitnami/kafka/templates/kafka-provisioning-serviceaccount.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.provisioning.serviceAccount.create }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "kafka.client.serviceAccountName" . }}
+  namespace: {{ template "zookeeper.namespace" . }}
```
Comment (jotamartos) [inline bitnami/kafka/templates/kafka-provisioning-serviceaccount.yaml:6]: namespace: {{ .Release.Namespace | quote }}

**Review by jotamartos — CHANGES_REQUESTED** [review]
Thanks for your contribution. Could you please take a look at my comments? Thanks

File: bitnami/kafka/Chart.yaml [hunk 4]
```diff
@@ -29,4 +29,4 @@ name: kafka
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/kafka
   - https://kafka.apache.org/
-version: 19.0.1
+version: 19.0.2
```
Comment (jotamartos) [inline bitnami/kafka/Chart.yaml:32]: Please bump the chart to a new minor version, you are adding new functionalities.

```
19.1.0
```

**Comment by cleverhu** [comment]
> Thanks for your contribution. Could you please take a look at my comments? Thanks

I will update the code later.Thanks

File: bitnami/kafka/values.yaml (hunk 1 above)
Comment (cleverhu) [reply bitnami/kafka/values.yaml:1752]: Should I remove the params for README?

**Review by jotamartos — APPROVED** [review]
Changes LGTM

**Pull request merged**